Lucas_Theorem/Lucas_Theorem.c

```diff
@@ -0,0 +1,55 @@
+// Lucas's theorem expresses the remainder of division of the binomial coefficient nCr
+// by a prime number p in terms of the base p expansions of the integers m and n.
+
+# include <stdio.h>
+# include <string.h>   
+
+int nCrModpDP (int n, int r, int p) ;
+int nCrModpLucas (int n, int r, int p) ;
+  
+int main() 
+{ 
+    int n , r , p ; 
+    printf ( "Enter the value of n (in nCr) : " ) ;
+    scanf ( "%d" , &n ) ;
+    printf ( "Enter the value of r (in nCr) : " ) ;
+    scanf ( "%d" , &r ) ;
+    printf ( "Enter the value of p: " ) ;
+    scanf ( "%d" , &p ) ;
+    
+    printf ( "Value of nCr mod %d is %d " , p , nCrModpLucas(n, r, p)); 
+    return 0; 
+}
+
+int nCrModpDP (int n, int r, int p) 
+{     
+    int C[r+1]; 
+    memset ( C, 0, sizeof(C) ); //memset() is used to fill a block of memory with a particular value.
+  
+    C[0] = 1; // Top row of Pascal Triangle 
+  
+    // One by constructs remaining rows of Pascal Triangle  
+    for (int i = 1; i <= n; i++) 
+    { 
+        int x = i ;
+        if ( i > r )
+            x = r ;
+        for (int j = x; j > 0; j--)   
```

Comment:
why can't directly use min(i, r) ?

Reply:
@sridharjajoo I have done the required change (using min). Please review.